java-checks/src/main/java/org/sonar/java/checks/LeastSpecificTypeCheck.java

```diff
@@ -164,21 +161,21 @@ private void computeChainsForSuperType(List<List<Type>> result, Symbol m, Type t
       }
     }
 
-    private static boolean definesOrInheritsSymbol(Symbol symbol, JavaSymbol.TypeJavaSymbol typeSymbol) {
+    private static boolean definesOrInheritsSymbol(Symbol symbol, Symbol.TypeSymbol typeSymbol) {
       return definesSymbol(symbol, typeSymbol)
-        || typeSymbol.superTypes().stream().anyMatch(superType -> definesSymbol(symbol, superType.symbol()));
+        || JUtils.superTypes(typeSymbol).stream().anyMatch(superType -> definesSymbol(symbol, superType.symbol()));
     }
 
     private static boolean definesSymbol(Symbol m, Symbol.TypeSymbol typeSymbol) {
-      return typeSymbol.memberSymbols().stream().anyMatch(s -> isOverriding(m, s, ((ClassJavaType) typeSymbol.type())));
+      return typeSymbol.memberSymbols().stream().anyMatch(s -> isOverriding(m, s));
```

Comment:
It was already like this before your changes, but must admit that implementation of this check looks to me either extremely inefficient or extremely redundant: `isOverriding` checks `m.isMethodSymbol` and `m` doesn't depend from elements of stream.